config/manager/manager.yaml

```diff
@@ -29,6 +29,7 @@ spec:
         args:
         - --leader-elect
         - --zap-log-level=$(ZAP_LOG_LEVEL)
+        - --MaxConcurrentReconciles=20
```

Comment:
can we extract this out to an env var (like `$(ZAP_LOG_LEVEL)` above) so that we can more easily configure this without needing to make manifest changes?

If that's outside the scope of this PR, maybe even just for now set the env with
```
- name: MAX_CONCURRENT_RECONCILES
  value: 20
```